src/modules/10-common/components/NGBreadcrumbs/__tests__/NGBreadcrumbsV2.test.tsx

```diff
@@ -93,4 +93,18 @@ describe('NGBreadcrumbsV2', () => {
       expect(screen.getByText('projectLabel: selectedProjectName')).toBeInTheDocument()
     })
   })
+  test('should invoke onClick passed in the link, when present', async () => {
+    const linkFn = jest.fn()
+    const { container } = render(
+      <TestWrapper>
+        <NGBreadcrumbsV2 links={[{ url: 'url', label: 'newlabel', onClick: linkFn }]} />
+      </TestWrapper>
+    )
+    fireEvent.click(container.querySelector('span[class="bp3-popover-target"]') as HTMLElement)
+    await waitFor(() => {
+      expect(screen.getByText('newlabel')).toBeInTheDocument()
+    })
+    fireEvent.click(screen.getByText('newlabel'))
+    expect(linkFn).toBeCalled()
+  })
```

Comment:
Some line breaks would make this _a lot_ easier to read. As-is it's a wall of text that people would need to purposefully read every line to understand.

In general it's a good idea to try to make "setup"/"action"/"expected outcome" section obvious.

Also, try to favour `userEvent` over `fireEvent` and you can save some complexity by using `findBy` instead of `waitFor` and `getBy`.

```suggestion

  test('should invoke onClick passed in the link, when present', async () => {
    const linkFn = jest.fn()
    render(
      <TestWrapper>
        <NGBreadcrumbsV2 links={[{ url: 'url', label: 'newlabel', onClick: linkFn }]} />
      </TestWrapper>
    )
    
    userEvent.click(document.querySelector('span[class="bp3-popover-target"]') as HTMLElement)
    userEvent.click(await screen.findByText('newlabel'))

    expect(linkFn).toHaveBeenCalled()
  })
```

Reply:
I read this after the commit. I will address in next PR